### PR TITLE
Fix WandBLogger to allow resuming runs with updated config values

### PR DESCRIPTION
**Pull request by parthsarthi03**
#### Context
This PR aims to:
- [x] fix a bug

It addresses the issue #1080 where the `WandBLogger` fails to resume a run with an updated config file when `allow_val_change` is set to `True` in the config.

#### Changelog
The changes made in this PR are:
1. Added `self.config_allow_val_change = kwargs.get('allow_val_change', False)` in the `__init__` method of the `WandBLogger` class to retrieve the value of `allow_val_change` from the `kwargs` dictionary and store it as an instance variable.
2. Modified the line `self._wandb.config.update(resolved)` to `self._wandb.config.update(resolved, allow_val_change=self.config_allow_val_change)` in the `log_config` method to pass the `allow_val_change` option to the `update` method based on the value provided during the initialization of the `WandBLogger`.

#### Test plan
 Tested the modified `WandBLogger` by resuming a run with an updated config file where `allow_val_change` was set to `True`. The run resumed successfully without any errors, and the config values were updated as expected.

### CODE REVIEW

**Comment by ebsmothers** [comment]
Thanks for the PR! This looks fine to me, will merge once CI is green.

**Comment by ebsmothers** [comment]
Thanks @parthsarthi03 for the PR! The change looks fine to me, but lint is failing. Mind fixing locally? Once CI is green I can merge this

**Comment by parthsarthi03** [comment]
> Thanks @parthsarthi03 for the PR! The change looks fine to me, but lint is failing. Mind fixing locally? Once CI is green I can merge this

Done!

**Review by RdoubleA — APPROVED** [review]
Thanks for the fix! CI is green, so will merge this in

**Pull request merged**
